src/runtime/program/vote/execute.zig

```diff
@@ -428,6 +429,61 @@ fn executeAuthorizeChecked(
     );
 }
 
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_processor.rs#L114-L118
+fn executeUpdateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+) InstructionError!void {
+    try ic.info.checkNumberOfAccounts(2);
+
+    var new_identity = try ic.borrowInstructionAccount(
+        @intFromEnum(vote_instruction.UpdateVoteIdentity.AccountIndex.new_identity),
+    );
+    defer new_identity.release();
+
+    try updateValidatorIdentity(
+        allocator,
+        ic,
+        vote_account,
+        new_identity.pubkey,
+    );
+}
+
+/// Agave https://github.com/anza-xyz/agave/blob/24e62248d7a91c090790e7b812e23321fa1f53b1/programs/vote/src/vote_state/mod.rs#L722
+///
+/// Update the node_pubkey, requires signature of the authorized voter
+fn updateValidatorIdentity(
+    allocator: std.mem.Allocator,
+    ic: *InstructionContext,
+    vote_account: *BorrowedAccount,
+    new_identity: Pubkey,
+) InstructionError!void {
+    const versioned_state = try vote_account.deserializeFromAccountData(
+        allocator,
+        VoteStateVersions,
+    );
+
+    var vote_state = versioned_state.convertToCurrent(allocator) catch {
+        // TODO okay to convert out of memory to InvalidAccountData?
+        return InstructionError.InvalidAccountData;
```

Comment:
should fix this

Reply:
[6b84b02](https://github.com/Syndica/sig/pull/611/commits/6b84b02e3962fe3bc35439ef2247851469eebeec)